### ISSUE TITLE
Will it work with DeepLabV3?

### ISSUE THREAD

**Issue by DanielZanchi** [cainxx/image-segmenter-ios#2]
Is it possible to have it working with DeepLabV3 model?

**Comment by cainxx** [comment]
It should be ok, I have not tested. ues coremltools  https://github.com/apple/coremltools

**Comment by DanielZanchi** [comment]
I have the Model (from Apple website), but the file UIImage+segmentation.swift needs ti ne readapted to use that model, some things have changed.

https://docs-assets.developer.apple.com/coreml/models/Image/ImageSegmentation/DeepLabV3/DeepLabV3.mlmodel

**Comment by ismaeIfm** [comment]
I adapted the code to use custom deeplabv3 converted with tfcoreml. You can see https://github.com/ismaeIfm/image-segmenter-ios

**Comment by l0stpenguin** [comment]
@ismaelfm is ths v3 model fast enough to be used for real time purposes?

**Comment by ismaeIfm** [comment]
@mevinDhun I don't think so, I've tested it on an iPhone XS and it takes a couple of seconds to perform the segmentation on the test image, but I'm not sure if some settings need to be changed to use the neural engine. I think a proper benchmarking is needed to answer properly.

**Comment by DanielZanchi** [comment]
> I adapted the code to use custom deeplabv3 converted with tfcoreml. You can see https://github.com/ismaeIfm/image-segmenter-ios

Why don't you use the one apple gives you instead of converting it?

https://docs-assets.developer.apple.com/coreml/models/Image/ImageSegmentation/DeepLabV3/DeepLabV3.mlmodel

**Comment by ismaeIfm** [comment]
Because the Apple coreml model is “compiled” from input to predictions and this code assumes the output of the model is an internal layer (ResizeBilinear), so in order to make it work the code would need some changes and I'm not a swift developer.
